src/lib.rs

```diff
@@ -234,16 +234,36 @@ impl Coord {
 
     #[allow(clippy::cast_possible_truncation, clippy::cast_possible_wrap)] // Impossible, since the RegEx limits length
     fn parse_component(val: &str) -> Result<f64, ()> {
-        let mut parts = val.split([':', '.']);
-        let deg = Self::parse_number_opt(parts.next())?;
-        let min = Self::parse_number_opt(parts.next())?;
-        let sec = Self::parse_number_opt(parts.next())?;
-        let mut total = f64::from(deg) + f64::from(min) / 60.0 + f64::from(sec) / 3600.0;
-        if let Some(fractional) = parts.next() {
-            let frac = fractional.parse::<u16>().map_err(|_| ())?;
-            total += f64::from(frac) / 10_f64.powi(fractional.len() as i32) / 3600.0
+        // Split by colon to separate degrees from minutes/seconds
+        let mut colon_parts = val.split(':');
+        let deg = Self::parse_number_opt(colon_parts.next())?;
+
+        // Get the second part (minutes or minutes:seconds)
+        let rest = colon_parts.next().ok_or(())?;
```

Comment:
This comment and the variable is not accurate, right? This will query the second component split by colon, i.e. the integer minutes in DMS format, or the decimal minutes in DDM format.

Suggestion:

```suggestion
        // Get the minutes (decimal if DDM format or integer if DMS format)
        let raw_minutes = colon_parts.next().ok_or(())?;
```